### PR TITLE
Optimize item transfer and inventory sorting

### PR DESCRIPTION
**Pull request by kuba6000**
Fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10002
Fixes https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/5490

### CODE REVIEW

**Comment by kuba6000** [comment]
Results:
Before:
image
After:
image

**Review by Glease — COMMENTED** [review]
Haven't checked everything. Will continue tomorrow.

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_InputBus.java [hunk 1]
```diff
@@ -150,16 +155,39 @@ public void updateSlots() {
...
-            for (int i = 0; i < mInventory.length; i++)
-                for (int j = i + 1; j < mInventory.length; j++)
-                    if (mInventory[j] != null && mInventory[j].stackSize <= 0 && (mInventory[i] == null || GT_Utility.areStacksEqual(mInventory[i], mInventory[j])))
-                        GT_Utility.moveStackFromSlotAToSlotB(getBaseMetaTileEntity(), getBaseMetaTileEntity(), j, i, (byte) 64, (byte) 1, (byte) 64, (byte) 1);
-        } else {
-            for (int i = 0; i < mInventory.length; i++)
-                for (int j = i + 1; j < mInventory.length; j++)
-                    if (mInventory[j] != null && (mInventory[i] == null || GT_Utility.areStacksEqual(mInventory[i], mInventory[j])))
-                        GT_Utility.moveStackFromSlotAToSlotB(getBaseMetaTileEntity(), getBaseMetaTileEntity(), j, i, (byte) 64, (byte) 1, (byte) 64, (byte) 1);
+        // no order, this is input bus :>
```
[the remark this answers is not in the document]
Comment (kuba6000) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Hatch_InputBus.java]: Ok, made it ordered

File: src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Buffer.java [hunk 2]
```diff
@@ -346,19 +348,37 @@ public void updateSlots() {
...
-                if (mInventory[j] != null && (mInventory[i] == null || GT_Utility.areStacksEqual(mInventory[i], mInventory[j])))
-                    GT_Utility.moveStackFromSlotAToSlotB(getBaseMetaTileEntity(), getBaseMetaTileEntity(), j, i, (byte) 64, (byte) 1, (byte) 64, (byte) 1);
-            }
+            validSlots.add(i);
+            ItemStack s = mInventory[i];
+            if(s == null)
+                continue;
+            int ol = s.stackSize;
+            s.stackSize = 1;
+            String sID = s.toString() + (s.hasTagCompound() ? s.getTagCompound().toString() : "");
```
[the remark this answers is not in the document]
Comment (kuba6000) [reply src/main/java/gregtech/api/metatileentity/implementations/GT_MetaTileEntity_Buffer.java]: I have made ItemID public and used it instead

**Pull request merged**